File: frontend/src/types/trip.ts
import type { ChatPlannerPhase } from "@/types/trip-conversation";

export type TripCreateRequest = {
  browser_session_id: string;
  title?: string;
};

export type TripCreateResponse = {
  trip_id: string;
  browser_session_id: string;
  thread_id: string;
  title: string;
  trip_status: "collecting_requirements";
  thread_status: "ready";
  created_at: string;
};

export type TripListItemResponse = TripCreateResponse & {
  updated_at: string;
  phase: ChatPlannerPhase | null;
  brochure_ready: boolean;
  latest_brochure_snapshot_id: string | null;
  latest_brochure_version: number | null;
  brochure_versions_count: number;
  from_location: string | null;
  to_location: string | null;
  start_date: string | null;
  end_date: string | null;
  travel_window: string | null;
  trip_length: string | null;
  selected_modules: string[];
  timeline_item_count: number;
};

export type TripListResponse = {
  items: TripListItemResponse[];
};

export type TripDeleteResponse = {
  trip_id: string;
  deleted: true;
};

File: backend/app/schemas/trip.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


TripStatus = Literal["collecting_requirements"]
ThreadStatus = Literal["ready"]


class TripCreateRequest(BaseModel):
    browser_session_id: str = Field(..., min_length=1, max_length=80)
    title: str | None = Field(default=None, min_length=1, max_length=160)


class TripCreateResponse(BaseModel):
    trip_id: str
    browser_session_id: str
    thread_id: str
    title: str
    trip_status: TripStatus
    thread_status: ThreadStatus
    created_at: datetime


class TripListItemResponse(TripCreateResponse):
    updated_at: datetime
    phase: str | None = None
    brochure_ready: bool = False
    latest_brochure_snapshot_id: str | None = None
    latest_brochure_version: int | None = None
    brochure_versions_count: int = 0
    from_location: str | None = None
    to_location: str | None = None
    start_date: str | None = None
    end_date: str | None = None
    travel_window: str | None = None
    trip_length: str | None = None
    selected_modules: list[str] = Field(default_factory=list)
    timeline_item_count: int = 0


class TripListResponse(BaseModel):
    items: list[TripListItemResponse]


class TripDeleteResponse(BaseModel):
    trip_id: str
    deleted: bool = True
